openmc/settings.py

```diff
@@ -1027,82 +1075,80 @@ def track(self, track):
 
         self._track = track
 
-
     @ufs_dimension.setter
     def ufs_dimension(self, dimension):
-
-        if not is_integer(dimension) and not is_float(dimension):
-            msg = 'Unable to set UFS dimension to non-integer or ' \
-                  'non-floating point value {0}'.format(dimension)
+        if not isinstance(dimension, (tuple, list)):
+            msg = 'Unable to set UFS mesh dimension to {0} which is ' \
+                  'not a Python tuple or list'.format(dimension)
             raise ValueError(msg)
 
-        elif dimension < 1:
-            msg = 'Unable to set UFS dimension to value {0} which is ' \
-                  'less than one'.format(dimension)
+        elif len(dimension) == 3:
+            msg = 'Unable to set UFS mesh dimension to {0} which is ' \
+                  'not a set of 3 integer dimensions'.format(dimension)
             raise ValueError(msg)
 
-        self._ufs_dimension = dimension
+        for dim in dimension:
+            if not is_integer(dim):
+                msg = 'Unable to set entropy mesh dimension to a ' \
+                      'non-integer {0}'.format(dim)
+                raise ValueError(msg)
+            elif dim < 1:
+                msg = 'Unable to set UFS dimension to value {0} which is ' \
+                      'less than one'.format(dimension)
+                raise ValueError(msg)
 
+        self._ufs_dimension = dimension
 
     @ufs_lower_left.setter
     def ufs_lower_left(self, lower_left):
-
         if not isinstance(lower_left, (tuple, list, np.ndarray)):
             msg = 'Unable to set UFS mesh lower left corner to {0} which is ' \
                   'not a Python tuple or list'.format(lower_left)
             raise ValueError(msg)
 
-        elif len(lower_left) < 3 or len(lower_left) > 3:
+        elif len(lower_left) == 3:
             msg = 'Unable to set UFS mesh lower left corner to {0} which ' \
                   'is not a 3D point'.format(lower_left)
             raise ValueError(msg)
 
         self._ufs_lower_left = lower_left
 
-
     @ufs_upper_right.setter
     def ufs_upper_right(self, upper_right):
-
-        if not isinstance(upper_right, tuple) and \
-          not isinstance(upper_right, list):
+        if not isinstance(upper_right, (tuple, list)):
             msg = 'Unable to set UFs mesh upper right corner to {0} which is ' \
                   'not a Python tuple or list'.format(upper_right)
             raise ValueError(msg)
 
-        if len(upper_right) < 3 or len(upper_right) > 3:
+        if len(upper_right) == 3:
```

Comment:
ditto ``!=`` rather than ``==`` (or perhaps i'm just tired and not thinking about this correctly)